lua/lazyvim/plugins/extras/coding/mini-snippets.lua

```diff
@@ -0,0 +1,159 @@
+        -- "snippets/" subdirectories from 'runtimepath' directories.
+        snippets.gen_loader.from_lang(), -- this is the default in the extra...
+      }
+    end,
+  },
+}
+
+--]]
+
+local function expand(args)
+  ---@diagnostic disable-next-line: undefined-global
+  local insert = MiniSnippets.config.expand.insert or MiniSnippets.default_insert
+  insert({ body = args.body }) -- insert at cursor
+end
+
+local snippet_select_for_cmp = function(snippets, insert, opts)
+  local cmp = require("cmp")
+  if cmp.visible() then
+    cmp.close()
+  end
+
+  MiniSnippets.default_select(snippets, insert, opts)
+end
+local snippet_select_for_blink = function(snippets, insert, opts)
+  -- Blink's cancel uses vim.schedule!
+  require("blink.cmp").cancel()
+
+  -- Schedule, otherwise blink's virtual text is not removed on vim.ui.select
+  vim.schedule(function()
+    MiniSnippets.default_select(snippets, insert, opts)
+  end)
+end
+local snippet_select_without_completion_engine = function(snippets, insert, opts)
+  MiniSnippets.default_select(snippets, insert, opts)
+end
+local snippet_select = snippet_select_without_completion_engine
+
+return {
+  -- disable builtin snippet support
+  { "garymjr/nvim-snippets", optional = true, enabled = false },
+  -- disable luasnip:
+  { "L3MON4D3/LuaSnip", optional = true, enabled = false },
+
+  -- add mini.snippets
+  desc = "Beta testing mini.snippets, a plugin to manage and expand snippets (alternative for luasnip)",
+  {
+    "echasnovski/mini.snippets",
+    event = "InsertEnter",
+    dependencies = {
+      {
+        "rafamadriz/friendly-snippets",
+      },
+    },
+    opts = function()
+      -- Unset snippet_forward. Handled by mini.snippets:
+      LazyVim.cmp.actions.snippet_forward = nil
+
+      ---Dummy snippet_stop, only needed because "esc" is overriden. See keymaps.lua
+      ---@diagnostic disable-next-line: duplicate-set-field
+      LazyVim.cmp.actions.snippet_stop = function() end
+
+      local snippets = require("mini.snippets")
+      return {
+        snippets = {
+          -- Load snippets based on current language by reading files from
+          -- "snippets/" subdirectories from 'runtimepath' directories.
+          snippets.gen_loader.from_lang(),
+        },
+        expand = {
+          select = function(...)
+            snippet_select(...) -- if needed, close completion windows on snippet select
+          end,
+        },
+      }
+    end,
+    config = function(_, opts)
+      if opts.mappings and (opts.mappings.jump_next or opts.mappings.jump_previous) then
+        LazyVim["warn"]({
+          "`mini.snippets`:",
+          "Don't override jump_next or jump_previous.",
+          "This does not work correctly in all cases.",
+        }, { title = "LazyVim" })
+      end
+      require("mini.snippets").setup(opts)
+    end,
+  },
+
+  -- nvim-cmp integration: Use mini.snippets to expand snippets from lsp
```

Comment:
I'd also assume that at least *some* people would want to have 'nvim-cmp' suggest snippets that 'mini.snippets' prepares/matches in current context. The array of snippets can be computed [like so](https://github.com/echasnovski/mini.nvim/blob/733afb94ed88d32b839a644dc137604691e46032/doc/mini-snippets.txt#L478-L484). When to do it and shaping into correct data structure is expected to be done on the "adapter's" side.

Reply:
> I'd also assume that at least some people would want to have 'nvim-cmp' suggest ...

I am sure of that, also regarding `blink.cmp`...)
Although, maybe some people see the benefit of separating non-lsp snippets from the completion engine. I do at least. 

1. Less clutter in the available completions
2. A more purposeful and direct approach of using snippets in the code. Typing `ll<c-j>` for example is a very fast way to define a `lua` variable.

This extra provides a quick start for `mini.snippets` in its current state. When sources for `nvim-cmp` or `blink.cmp` become available, `mini.snippets` would likely be connected to the completion engine in the same manner as seen in the [luasnip](https://github.com/LazyVim/LazyVim/blob/d0c366e4d861b848bdc710696d5311dca2c6d540/lua/lazyvim/plugins/extras/coding/luasnip.lua#L31) extra. 
Also, when the snippets are offered by the completion engine, there could be a way to disable `vim.ui.select` from triggering on `expand`?

For now, I thought it best to only implement the "expand" part, and let other parts be completely handled by `mini.snippets`. 

Do you agree? Maybe you have a suggestion to modify the comment at the start of the `extra` to be more clear.

Is it correct that you are considering an `lsp` solution to expose `mini.snippets` to the completion engine? In that case, separate sources for `blink.cmp` or `nvim-cmp` might not be needed. One would lose the ability to prioritize `sources` though.

Reply:
> Also, when the snippets are offered by the completion engine, there could be a way to disable `vim.ui.select` from triggering on `expand`?

There are four steps for users/config/distro to check that, so probably possible.

> For now, I thought it best to only implement the "expand" part, and let other parts be completely handled by `mini.snippets`.
> 
> Do you agree? Maybe you have a suggestion to modify the comment at the start of the `extra` to be more clear.

I don't have many opinions here, to be honest. Just that there certainly are people who prefer to have snippets as completion suggestions even if they don't come from LSP server. I am not one of those people. What's interesting for me is to make sure that 'mini.snippets' has all it needs to export for this to be implemented.

> Is it correct that you are considering an `lsp` solution to expose `mini.snippets` to the completion engine?

Yes, this is one of the next goals together with 'mini.completion' supporting LSP snippets.

Reply:
> I don't have many opinions here, to be honest. Just that there certainly are people who prefer to have snippets as completion suggestions even if they don't come from LSP server. I am not one of those people. What's interesting for me is to make sure that 'mini.snippets' has all it needs to export for this to be implemented.

I'm very sorry for not understanding the statement above completely... Would you like me to assign the `draft` status to this PR, perhaps until those sources have been written?
For `nvim-cmp`, something similar as `cmp_ luasnip`(written by the author of `luasnip`). For `blink.cmp` there is already an [open](https://github.com/Saghen/blink.cmp/issues/741) issue.

Reply:
> I'm very sorry for not understanding the statement above completely... Would you like me to assign the `draft` status to this PR, perhaps until those sources have been written?
> For `nvim-cmp`, something similar as `cmp_ luasnip`(written by the author of `luasnip`). For `blink.cmp` there is already an [open](https://github.com/Saghen/blink.cmp/issues/741) issue.

Please don't. It was a drive-by comment describing what should be possible and might be valuable for some people. I think having only 'mini.snippets' `insert` functionality used as snippet engine for other completion engines is very valuable in itself. Thanks for making this :+1:

Reply:
> Thanks for making this

It's my pleasure!
Could you have a look at the latest commit(a0885fc3f1f7)? I changed the comment in the first block a bit to hopefully express the current state of the extra more clearly.

Reply:
> Could you have a look at the latest commit([a0885fc](https://github.com/LazyVim/LazyVim/commit/a0885fc3f1f76ef7dc00c669f882347bada26c44))? I changed the comment in the first block a bit to hopefully express the current state of the extra more clearly.

Sure, looks good. I don't think it *has* to be this detailed and formal for Folke, but having extra clarity for him to trim down from might be indeed useful.

Reply:
Thanks for the review!